packages/kbn-text-based-editor/src/text_based_languages_editor.tsx

```diff
@@ -55,154 +40,63 @@ import { addQueriesToCache, updateCachedQueries } from './history_local_storage'
-   */
-  detectedTimestamp?: string;
-  /** Array of errors */
-  errors?: Error[];
-  /** Warning string as it comes from ES */
-  warning?: string;
-  /** Disables the editor and displays loading icon in run button
-   * It is also used for hiding the history component if it is not defined
-   */
-  isLoading?: boolean;
-  /** Disables the editor */
-  isDisabled?: boolean;
-  /** Indicator if the editor is on dark mode */
-  isDarkMode?: boolean;
-  dataTestSubj?: string;
-  /** If true it hides the minimize button and the user can't return to the minimized version
-   * Useful when the application doesn't want to give this capability
-   */
-  hideMinimizeButton?: boolean;
-  /** Hide the Run query information which appears on the footer*/
-  hideRunQueryText?: boolean;
-  /** This is used for applications (such as the inline editing flyout in dashboards)
-   * which want to add the editor without being part of the Unified search component
-   * It renders a submit query button inside the editor
-   */
-  editorIsInline?: boolean;
-  /** Disables the submit query action*/
-  disableSubmitAction?: boolean;
-
-  /** when set to true enables query cancellation **/
-  allowQueryCancellation?: boolean;
-
-  /** hide @timestamp info **/
-  hideTimeFilterInfo?: boolean;
-
-  /** hide query history **/
-  hideQueryHistory?: boolean;
-
-  /** hide header buttons when editor is expanded */
-  hideHeaderWhenExpanded?: boolean;
-}
-
-interface TextBasedEditorDeps {
-  core: CoreStart;
-  dataViews: DataViewsPublicPluginStart;
-  expressions: ExpressionsStart;
-  indexManagementApiService?: IndexManagementPluginSetup['apiService'];
-  fieldsMetadata?: FieldsMetadataPublicStart;
-}
-
-const MAX_COMPACT_VIEW_LENGTH = 250;
-const FONT_WIDTH = 8;
-const EDITOR_ONE_LINER_UNUSED_SPACE = 180;
-const EDITOR_ONE_LINER_UNUSED_SPACE_WITH_ERRORS = 220;
-
 const KEYCODE_ARROW_UP = 38;
 const KEYCODE_ARROW_DOWN = 40;
 
 // for editor width smaller than this value we want to start hiding some text
 const BREAKPOINT_WIDTH = 540;
 
-let clickedOutside = false;
-let initialRender = true;
-let updateLinesFromModel = false;
-let lines = 1;
-let isDatePickerOpen = false;
-
 export const TextBasedLanguagesEditor = memo(function TextBasedLanguagesEditor({
   query,
   onTextLangQueryChange,
   onTextLangQuerySubmit,
-  expandCodeEditor,
-  isCodeEditorExpanded,
   detectedTimestamp,
   errors: serverErrors,
   warning: serverWarning,
   isLoading,
   isDisabled,
-  isDarkMode,
-  hideMinimizeButton,
   hideRunQueryText,
   editorIsInline,
   disableSubmitAction,
   dataTestSubj,
   allowQueryCancellation,
   hideTimeFilterInfo,
   hideQueryHistory,
-  hideHeaderWhenExpanded,
+  isHelpMenuOpen,
+  setIsHelpMenuOpen,
 }: TextBasedLanguagesEditorProps) {
   const popoverRef = useRef<HTMLDivElement>(null);
+  const datePickerOpenStatusRef = useRef<boolean>(false);
   const { euiTheme } = useEuiTheme();
-  const language = getAggregateQueryMode(query);
```

Comment:
I removed that, there is only one language to rule them all. Also makes the code cleaner and no darkMode dependency 😉